src/app/modules/library/library-project-menu/library-project-menu.component.ts

```diff
@@ -38,6 +41,7 @@ export class LibraryProjectMenuComponent {
     this.isCanShare = this.isOwner() && !this.isRun;
     this.editLink = `${this.configService.getContextPath()}/teacher/edit/unit/${this.project.id}`;
     this.isChild = this.project.isChild();
+    this.archived = this.project.tags.includes('archived');
```

Comment:
Encapsulate ```Project.tags``` field by defining ```Project.hasTag()```?
```suggestion
    this.archived = this.project.hasTag('archived');
```